### PR TITLE
 Configurable remote diarization server timeout

### PR DESCRIPTION
**Pull request by michalblaha**
via `.env` file or environment variable 
based on previous pull request, Modified according to proposals @aleksandr-smechov and @karmi
 
added patch version change

### CODE REVIEW

File: .env
Comment (aleksandr-smechov) [inline .env:1]: The version can remain the same, as it's a minor change

**Comment by aleksandr-smechov** [comment]
@michalblaha Left comments, thank you

File: src/wordcab_transcribe/services/asr_service.py [hunk 1]
```diff
@@ -776,7 +777,10 @@ async def remote_diarization(
         data: DiarizationRequest,
     ) -> DiarizationOutput:
         """Remote diarization method."""
-        async with aiohttp.ClientSession() as session:
+        # Set the timeout according to the env value of the REMOTE_DIARIZE_SERVER_REQUEST_TIMEOUT_SEC variable
+
+        specifictimeout = aiohttp.ClientTimeout(total=int(os.getenv("REMOTE_DIARIZE_SERVER_REQUEST_TIMEOUT_SEC", 300)))
```
[the remark this answers is not in the document]
Comment (michalblaha) [reply src/wordcab_transcribe/services/asr_service.py:782]: Hi.
1) .env - its there. See https://github.com/Wordcab/wordcab-transcribe/pull/283/files
2) config & Settings class - ok, I'll change it
3) I've tested it on my infrastructure in docker, with `-e "REMOTE_DIARIZE_SERVER_REQUEST_TIMEOUT_SEC=30000"`

**Comment by michalblaha** [comment]
Thanks, so do I. 
BTW, could you check annoying bug #280 ? thx

File: .env [hunk 2]
```diff
@@ -8,7 +8,7 @@
 # The name of the project, used for API documentation.
 PROJECT_NAME="Wordcab Transcribe"
 # The version of the project, used for API documentation.
-VERSION="0.5.1"
+VERSION="0.5.2.1"
```
Comment (aleksandr-smechov) [inline .env:11]: No version increment

File: .env [hunk 3]
```diff
@@ -115,4 +115,6 @@ TRANSCRIBE_SERVER_URLS=
 # e.g. SERVER_URLS="http://1.2.3.4:8000,http://4.3.2.1:8000"
 DIARIZE_SERVER_URLS=
 #
+# Timeout for connection and waiting for result from remote diarize server. 300 sec is default value
+REMOTE_DIARIZE_SERVER_REQUEST_TIMEOUT_SEC=300
```
Comment (aleksandr-smechov) [inline .env:119]: Just `REMOTE_DIARIZATION_SERVER_REQUEST_TIMEOUT=300`

to keep things consistent

File: src/wordcab_transcribe/__init__.py [hunk 4]
```diff
@@ -19,4 +19,4 @@
 # and limitations under the License.
 """Main module of the Wordcab ASR API."""
 
-__version__ = "0.5.2"
+__version__ = "0.5.2.1"
```
Comment (aleksandr-smechov) [inline src/wordcab_transcribe/__init__.py:22]: No increment

File: src/wordcab_transcribe/config.py [hunk 5]
```diff
@@ -272,4 +274,5 @@ def __post_init__(self):
     # Remote servers configuration
     transcribe_server_urls=transcribe_server_urls,
     diarize_server_urls=diarize_server_urls,
+    remote_diarization_request_timeout=int(os.getenv("REMOTE_DIARIZE_SERVER_REQUEST_TIMEOUT_SEC", 300)),
```
Comment (aleksandr-smechov) [inline src/wordcab_transcribe/config.py:277]: remote_diarization_request_timeout=getenv("REMOTE_DIARIZATION_SERVER_REQUEST_TIMEOUT", 300),

File: src/wordcab_transcribe/services/asr_service.py [hunk 6]
```diff
@@ -300,6 +301,9 @@ def __init__(
                 "You provided URLs for remote diarization server, no local model will"
                 " be used."
             )
+            logger.info(
+                f"remote diarization server timeout is {settings.remote_diarization_request_timeout}"
```
Comment (aleksandr-smechov) [inline src/wordcab_transcribe/services/asr_service.py:305]: `f"Remote diarization server timeout set to {settings.remote_diarization_request_timeout}"`

File: src/wordcab_transcribe/services/asr_service.py [hunk 7]
```diff
@@ -776,7 +780,11 @@ async def remote_diarization(
         data: DiarizationRequest,
     ) -> DiarizationOutput:
         """Remote diarization method."""
-        async with aiohttp.ClientSession() as session:
+        # Set the timeout according to the env value of the REMOTE_DIARIZE_SERVER_REQUEST_TIMEOUT_SEC variable
+        logger.info(f"settings.remote_diarization_request_timeout={settings.remote_diarization_request_timeout} secs")
```
Comment (aleksandr-smechov) [inline src/wordcab_transcribe/services/asr_service.py:784]: Can remove lines 783-785

**Review by aleksandr-smechov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**